enterprise/server/invocation_stat_service/invocation_stat_service.go

```diff
@@ -104,6 +104,10 @@ func (i *InvocationStatService) GetTrend(ctx context.Context, req *inpb.GetTrend
 
 	if role := req.GetQuery().GetRole(); role != "" {
 		q.AddWhereClause("role = ?", role)
+	} else {
+		// Don't include workflow invocations in trends since workflows just "wrap"
+		// other invocations, and we want to avoid double-counting.
+		q.AddWhereClause(`role != "CI_RUNNER"`)
```

Comment:
Can the client whitelist what it wants from the request instead? It's not nice to have the server silently messing with this query on an API that makes no mention of it / really shouldn't.

(Also, what is the set of roles we currently know about?)

Separately, we should profile this query to ensure it works for our customers with millions of invocations.

Reply:
Updated for the client to send a whitelist of roles.

Did some profiling, for our largest customer it takes 5.8s for this query _without_ any `role = ?` clauses, and around 7.3s for this query with `role = ?` clauses (including our existing `role = "CI"` clause).

Worked with Tyler offline, added a new index which includes the role, and now both types of queries (with or without "role" filters) are just as fast.